pkg/secrets/check_rights_nix.go

```diff
@@ -13,15 +13,23 @@ import (
 	"syscall"
 )
 
-func checkRights(path string) error {
+func checkRights(path string, options checkRightOptions) error {
 	var stat syscall.Stat_t
 	if err := syscall.Stat(path, &stat); err != nil {
 		return fmt.Errorf("invalid executable '%s': can't stat it: %s", path, err)
 	}
 
 	// checking that group and others don't have any rights
-	if stat.Mode&(syscall.S_IRWXG|syscall.S_IRWXO) != 0 {
-		return fmt.Errorf("invalid executable '%s', 'groups' or 'others' have rights on it", path)
+	var unwantedPermissions uint = syscall.S_IRWXO
```

Comment:
Why not:
```suggestion
	unwantedPermissions := syscall.S_IRWXO
```

Reply:
when I did that, the linter was failing in some ci build. because the syscall.S_IRWXO was define as different type.